packages/rum-core/src/common/apm-server.js

```diff
@@ -284,44 +243,46 @@ class ApmServer {
     })
   }
 
-  _send(data = [], type = 'transactions') {
-    if (data.length === 0) {
+  sendEvents(events) {
+    if (events.length === 0) {
+      return
+    }
+    const transactions = []
+    const errors = []
+    for (const event of events) {
+      if (event[TRANSACTIONS]) {
+        transactions.push(event[TRANSACTIONS])
+      }
+      if (event[ERRORS]) {
+        errors.push(event[ERRORS])
+      }
+    }
+    if (transactions.length === 0 && errors.length === 0) {
       return
     }
 
-    const payload = { [type]: data }
+    const payload = {
+      [TRANSACTIONS]: transactions,
+      [ERRORS]: errors
+    }
     const filteredPayload = this._configService.applyFilters(payload)
     if (!filteredPayload) {
       this._loggingService.warn('Dropped payload due to filtering!')
       return
     }
 
-    let ndjson
-    if (type === 'errors') {
-      ndjson = this.ndjsonErrors(filteredPayload[type])
-    } else if (type === 'transactions') {
-      ndjson = this.ndjsonTransactions(filteredPayload[type])
-    } else {
-      if (__DEV__) {
-        this._loggingService.debug('Dropped payload due to unknown data type')
-      }
-      return
-    }
-
+    let ndjson = []
     const metadata = this.createMetaData()
-    ndjson.unshift(NDJSON.stringify({ metadata }))
+    ndjson.push(NDJSON.stringify({ metadata }))
+
+    ndjson = ndjson.concat(
+      this.ndjsonErrors(filteredPayload[ERRORS]),
+      this.ndjsonTransactions(filteredPayload[TRANSACTIONS])
```

Comment:
Does the order of these events matter from the APM server perspective?

One area we should check is that since APM server sets all the timestamps for a transaction and it's spans in the same request, would that still work?

Reply:
Order does not matter as far as i checked with the server team. Can you confirm @simitt ? 

> One area we should check is that since APM server sets all the timestamps for a transaction and it's spans in the same request, would that still work?

I didn't get this question. Isn't the same even before as well when we had separate queues?

Reply:
Previously we sent the transaction with all of its spans in the same request, which means the APM Server might check for the first event and if that's a transaction then set the timestamp of all the following spans to the same value, but now the first event might be an error. This might not be how the APM server handles requests but I want to make sure the corner cases we have for the RUM payload will still work when we make this change.

Reply:
Current APM Server behavior is that the server records a `requestTime := time.Now` when the request comes in. If a span does not contain a dedicated timestamp, the spans timestamp is set to `requestTime + event.Start`. 

I am not certain if and how this behavior will change with the RUM specific optimizations @jalvz and @vigneshshanmugam are working on.

Reply:
Thanks for chiming in, We are not planning to change that behaviour with the RUM payload optimisations AFAIK. @jalvz Can you confirm that the optimisations that we do for the payload wont affect the ordering of events in any way?

Reply:
Yes, that's correct. However:

> Previously we sent the transaction with all of its spans in the same request

This is no longer the case?

Reply:
It's still the same, spans will follow transactions as they were before. This change only introduces error events inside the same payload, so it would look like this
```
<metadata>
<error 1>
<error 2>
<transaction1>
<spans of transaction 1>
<metricsets 1 if available>
<transaction2>
<spans of transaction 2>
<metricsets 2 if available>
```